File: app/src/main/java/com/fittrack/data/entity/PersonalRecord.kt
package com.fittrack.data.entity

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(
    tableName = "personal_records",
    foreignKeys = [
        ForeignKey(
            entity = Exercise::class,
            parentColumns = ["id"],
            childColumns = ["exerciseId"],
            onDelete = ForeignKey.CASCADE
        )
    ],
    indices = [Index("exerciseId")]
)
data class PersonalRecord(
    @PrimaryKey(autoGenerate = true) val id: Int = 0,
    val exerciseId: Int,
    val userId: Int,
    val recordType: String, // "max_weight", "max_reps", "max_volume", "est_1rm"
    val value: Double,
    val weight: Double = 0.0,
    val reps: Int = 0,
    val workoutSessionId: String = "",
    val achievedAt: Long = System.currentTimeMillis()
)

File: ios/P3Fitness/Models/PersonalRecord.swift
import SwiftData
import Foundation

@Model
final class PersonalRecord {
    var id: Int
    var exerciseId: Int
    var userId: Int
    var recordType: String // "max_weight", "max_reps", "max_volume", "est_1rm"
    var value: Double
    var weight: Double
    var reps: Int
    var workoutSessionId: String
    var achievedAt: Date

    init(exerciseId: Int, userId: Int, recordType: String, value: Double, weight: Double, reps: Int, workoutSessionId: String) {
        self.id = Int(Date().timeIntervalSince1970 * 1000000) % 1_000_000_000
        self.exerciseId = exerciseId
        self.userId = userId
        self.recordType = recordType
        self.value = value
        self.weight = weight
        self.reps = reps
        self.workoutSessionId = workoutSessionId
        self.achievedAt = Date()
    }
}
